README.md

```diff
@@ -101,249 +142,725 @@ The ```.url()``` method accepts the following parameters.
+| overlay_text_inner_alignment  | otia                            |
+| overlay_radius                | or                              |
+| progressive                   | pr                              |
+| lossless                      | lo                              |
+| trim                          | t                               |
+| metadata                      | md                              |
+| color_profile                 | cp                              |
+| default_image                 | di                              |
+| dpr                           | dpr                             |
+| effect_sharpen                | e-sharpen                       |
+| effect_usm                    | e-usm                           |
+| effect_contrast               | e-contrast                      |
+| effect_gray                   | e-grayscale                     |
+| original                      | orig                            |
+| raw                           | replaced by the parameter value |
 
 ## File Upload
 
 The SDK provides a simple interface using the `.upload_file()` method to upload files to the ImageKit Media library. It
-accepts all the parameters supported by the [ImageKit Upload API](https://docs.imagekit.io/api-reference/upload-file-api/server-side-file-upload).
+accepts all the parameters supported by
+the [ImageKit Upload API](https://docs.imagekit.io/api-reference/upload-file-api/server-side-file-upload).
 
-The `upload_file()` method requires at least the `file` and the `file_name` parameter to upload a file and returns a Dict with error or success data. Use `options` parameter to pass other parameters supported by the [ImageKit Upload API](https://docs.imagekit.io/api-reference/upload-file-api/server-side-file-upload). Use the same parameter name as specified in the upload API documentation.
+The `upload_file()` method requires at least the `file` as (URL/Base64/Binary) and the `file_name` parameter to upload a
+file. The method returns a Dict data in case of success, or it will throw a custom exception in case of failure.
+Use the `options` parameter to pass other parameters supported by
+the [ImageKit Upload API](https://docs.imagekit.io/api-reference/upload-file-api/server-side-file-upload). Use the same
+parameter name as specified in the upload API documentation.
 
 Simple usage
 
 ```python
-imagekit.upload_file(
-    file= "<url|base_64|binary>", # required
-    file_name= "my_file_name.jpg", # required
-    options= {
-        "folder" : "/example-folder/",
+import json
+
+result = imagekit.upload_file(
+    file="<url|base_64|binary>",  # required
+    file_name="my_file_name.jpg",  # required
+    options={
+        "folder": "/example-folder/",
         "tags": ["sample-tag"],
         "is_private_file": False,
         "use_unique_file_name": True,
         "response_fields": ["is_private_file", "tags"],
+        "extensions": json.dumps(
```

Comment:
Repeat mistake from Java. Why does the responsibility of converting JSON to string is on developer using the SDK and not the SDK itself?

Reply:
@imagekitio we hadn't changed this in Java as well. The reason for this being that there are too many types to cover so defining well defined types for this will require a lot of changes. Our goal for python SDK has been to bring it on parity with Java.